haystack/backends/solr_backend.py

```diff
@@ -392,6 +392,14 @@ def _process_results(self, raw_results, highlight=False, result_class=None, dist
             elif len(raw_results.spellcheck.get('suggestions', [])):
                 spelling_suggestion = raw_results.spellcheck['suggestions'][-1]
 
+            if isinstance(spelling_suggestion, dict):
+                # Solr 5+ JSON response format
+                if isinstance(spelling_suggestion.get('suggestion', [None])[-1], dict):
```

Comment:
you might be able to clean this up a bit by pulling the data out once into a new var, or overriding `spelling_suggestion` and using that.

foo = spelling_suggestion.get('suggestion', [None])[-1]
if isinstance(foo, dict):
 ...